### ISSUE TITLE
a11y attributes like aria-label and tabIndex should be in the VictoryBar instead of Bar component

### ISSUE THREAD

**Issue by DaoDaoNoCode** [FormidableLabs/victory#1746]
# Bugs and Questions

As above. For example, if we want to create accessible bar chart with default bar style, we still need to create a `<Bar>` component with only accessibility attributes like `aria-label` or `tabIndex` and pass the whole component to `dataComponent` prop in `<VictoryBar>`. However, these attributes should be in the `VictoryBar` component.

### Checklist

- [x] This is not a `victory-native` specific issue. (Issues that only appear in `victory-native` should be opened [here](https://github.com/FormidableLabs/victory-native/issues/new))

- [x] I have read through the [FAQ](https://formidable.com/open-source/victory/docs/faq) and [Guides](https://formidable.com/open-source/victory/guides/) before asking a question

- [x] I am using the latest version of Victory

- [x] I've searched open issues to make sure I'm not opening a duplicate issue

# Feature Requests

### Checklist

- [x] I've read through the [Docs](https://formidable.com/open-source/victory/docs) and [Guides](https://formidable.com/open-source/victory/guides) to make sure this functionality doesn't already exist

- [x] I've searched open issues to make sure I'm not opening a duplicate issue

### Description

The way to create bar chart now:
```
<VictoryBar
  style={{ data: { fill: "#c43a31" } }}
  data={accessibilityBarData}
  dataComponent={
     <Bar
        ariaLabel={({ datum }) => `x: ${datum.x}`}
        tabIndex={({ index }) => index + 1}
      />
  }
/>
```
The way I expected:
```
<VictoryBar
  style={{ data: { fill: "#c43a31" } }}
  data={accessibilityBarData}
  ariaLabel={`x: ${accessibilityBarData.x}`}
  tabIndex={0}
/>
```

**Comment by boygirl** [comment]
@DaoDaoNoCode We actually thought about doing this, and decided against it. It makes a ton of sense for components like `VictoryBar` that have a single, standard data component, but it gets really confusing when you think about components like `VictoryBoxPlot` that render several subcomponents that can be individually styled and labeled. To keep everything consistent, we opted to restrict the `ariaLabel` and `tabIndex` props to the primitive components rather than letting them be controlled by more complex components. I think the inconvenience of needing to define these props on the `dataComponent` is worth if for consistency across all components.
